.github/actions/load-workflow-variables/action.yml

```diff
@@ -0,0 +1,108 @@
+    default: 'env.yml'
+  find_in_parents:
+    description: 'If true, search for env_filename in parents until one is found.'
+    required: false
+    default: 'false'
+  top_level_directory:
+    description: 'The top level directory to stop in when inheriting environment files.'
+    required: false
+    default: '.'
+
+runs:
+  using: 'composite'
+  steps:
+    - name: 'Load From Local File Only'
+      if: |-
+        !contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.find_in_parents)
+      shell: 'bash'
+      working-directory: '${{ inputs.working_directory }}'
+      env:
+        WORKING_DIRECTORY: '${{ inputs.working_directory }}'
+        ENV_FILENAME: '${{ inputs.env_filename }}'
+        TOP_LEVEL_DIRECTORY: '${{ inputs.top_level_directory || github.workspace }}'
+      run: |-
+        WORKING_DIRECTORY=$(realpath "${WORKING_DIRECTORY}")
+
+        if [[ ! -f "$ENV_FILENAME" ]]; then
+            echo "ERROR: Failed to find target file ${ENV_FILENAME} in ${WORKING_DIRECTORY}"
+            exit 1
+        fi
+
+        PROPS=$(yq -o=props '.. | select(tag != "!!map" and tag != "!!seq") | ( (path | join("_")) + "<<EOT\n" + . + "\nEOT")' "${ENV_FILENAME}")
+
+        for PROP in "${PROPS}"; do
+            echo "${PROP}" >> $GITHUB_ENV
+        done
+
+    - name: 'Load From Local Or Parents'
+      if: |-
+        contains(fromJSON('["true", "True", "TRUE", "1", "T", "t"]'), inputs.find_in_parents)
+      shell: 'bash'
+      working-directory: '${{ inputs.working_directory }}'
+      env:
+        WORKING_DIRECTORY: '${{ inputs.working_directory }}'
+        ENV_FILENAME: '${{ inputs.env_filename }}'
+        TOP_LEVEL_DIRECTORY: '${{ inputs.top_level_directory || github.workspace }}'
+      run: |-
+        WORKING_DIRECTORY=$(realpath "${WORKING_DIRECTORY}")
+        TOP_LEVEL_DIRECTORY=$(realpath "${TOP_LEVEL_DIRECTORY}")
+
+        if [[ "${WORKING_DIRECTORY}" != "${TOP_LEVEL_DIRECTORY}"* ]]; then
+          echo "ERROR: ${TOP_LEVEL_DIRECTORY} is not a sub-directory of ${WORKING_DIRECTORY}"
+          exit 1
+        fi
+
+        echo -n "Checking ${WORKING_DIRECTORY} for ${ENV_FILENAME}..."
+        if [[ ! -f "$ENV_FILENAME" ]]; then
+          echo "MISSING"
+
+          while [[ "${WORKING_DIRECTORY}" != "/"  ]]; do
+            WORKING_DIRECTORY=$(dirname "${WORKING_DIRECTORY}")
+            cd "${WORKING_DIRECTORY}"
+
+            echo -n "Checking ${WORKING_DIRECTORY} for ${ENV_FILENAME}..."
+            if [ -f "${ENV_FILENAME}" ]; then
+              echo "FOUND"
+              break
+            fi
+
+            echo "MISSING"
+
+            if [[ "${WORKING_DIRECTORY}" == "${TOP_LEVEL_DIRECTORY}" ]]; then
+              echo "ERROR: Failed to find target file ${ENV_FILENAME}"
+              exit 1
+            fi
+          done
+        fi
+
+        PROPS=$(yq -o=props '.. | select(tag != "!!map" and tag != "!!seq") | ( (path | join("_")) + "<<EOT\n" + . + "\nEOT")' "${ENV_FILENAME}")
```

Comment:
Had the same thought, but in the future i thought it could be a feature to inherit all env files all the way up, so we would process this inside the loop.